src/map/clif.c

```diff
@@ -19097,6 +19098,381 @@ unsigned short clif_parse_cmd_optional( int fd, struct map_session_data *sd ) {
 }
 
 /*==========================================
+ * RoDEX
+ *------------------------------------------*/
+
+void clif_parse_rodex_open_write_mail(int fd, struct map_session_data *sd)
+{
+	const struct PACKET_CZ_REQ_OPEN_WRITE_MAIL *rPacket = RFIFOP(fd, 0);
+	struct PACKET_ZC_ACK_OPEN_WRITE_MAIL *sPacket = NULL;
+	WFIFOHEAD(fd, sizeof(*sPacket));
+	sPacket = WFIFOP(fd, 0);
+	sPacket->PacketType = rodexopenwrite;
+	strncpy(sPacket->receiveName, rPacket->receiveName, sizeof(sPacket->receiveName));
+	sPacket->result = (rodex->isenabled() == true) ? 1 : 0;
+	WFIFOSET(fd, sizeof(*sPacket));
+}
+
+void clif_parse_rodex_add_item(int fd, struct map_session_data *sd)
+{
+	struct PACKET_ZC_ADD_ITEM_TO_MAIL *packet;
+	const struct PACKET_CZ_ADD_ITEM_TO_MAIL *rPacket = RFIFOP(fd, 0);
+	int16 idx = rPacket->index - 2;
+	int j;
+
+	WFIFOHEAD(fd, sizeof(*packet));
```

Comment:
each packet must be sent from own function. This mean extract into other function sending packet